### PR TITLE
Merged the 2 active PRs (Unify Alerts / CoachView / Promote, Demote, Remove, Ban)

### PR DESCRIPTION
**Pull request by FrankreedX**
The 2 PR's now merged into this one:
- https://github.com/Golf-Drill-Challenge-App/Golf-App/pull/232
- https://github.com/Golf-Drill-Challenge-App/Golf-App/pull/190

### For testing (just in case anyone needed instructions):
- Run the app on 2 devices (e.g. emulator + actual phone)
- On one device log in as admin (owner), on another device sign in or sign up as a regular (player) user
- On admin device, try to remove a user (bug seems to trigger here in past). Try promote / demote / blacklist (ban) too

### CODE REVIEW

**Review by solderq35 — CHANGES_REQUESTED** [review]
## Testing
Issues with team tab > player profile:
- Promotion and Demotion cache invalidation is not working (coach pov). Need to do pulldown refresh on team tab index page
  - Should promote or demote player navigate the coach back to the team index page, similar to remove or ban player actions?
- I think pulldown refresh isn't working anymore for team tab > player profile (both coach and player pov)
- Ban user cache invalidation not working (from coach POV. Need to refresh team tab index page)
  - image
  - Note: Remove user does invalidate cache for team index page, from coach pov
- I sometimes got the error below but I couldn't reliably reproduce it, not sure why (for player removal or ban)
  - I think fixing the cache invalidation issues might fix this one? Needs more testing
  - image

**Review by solderq35 — COMMENTED** [review]
I think it mostly looks good now (promote / demote / ban working now with cache invalidation), but there are a few "alert" left in chooseTeam.js and input.js, I can try to resolve them real quick

**Comment by solderq35** [comment]
## Changes (https://github.com/Golf-Drill-Challenge-App/Golf-App/pull/239/commits/5073b08a8f305a0a737da6fdf423dd3f708cfc0f)
- Minor syntax changes:
  - Migrated remaining `alert` to `showDialog` syntax
  - Changed `console.errror` to `console.log(e)` + `showDialog`
- Make sure errors are handled for newly added functionality in remove user, blacklist (ban) user, promote / demote user, reset leaderboard (season)
- Remove outdated part of README about login
- Updated some of the "Document Updated" logs to be more specific

## Screenshots
image
image
image
image

**Review by hannacol — CHANGES_REQUESTED** [review]
Got an error about "Rendered fewer hooks than expected" when removing a player once, but did not happened again on a new user. All other promote, demote, and ban features work as intended

**Comment by solderq35** [comment]
> Got an error about "Rendered fewer hooks than expected" when removing a player once, but did not happened again on a new user. All other promote, demote, and ban features work as intended

Maybe will try this library later to help debug it: https://www.npmjs.com/package/eslint-plugin-react-hooks

**Comment by solderq35** [comment]
Added [eslint react-hooks](https://www.npmjs.com/package/eslint-plugin-react-hooks) for now as debug (see comments of `.eslintrc.js` file for more details).

**Commands:**
- `yarn lint` (runs `yarn eslint .`, which checks all files)
- `yarn eslint ./hooks` (for more control over which files get checked, in this case checking `hooks` directory)

File: app/content/profile/index.js [hunk 1]
```diff
@@ -340,7 +353,7 @@ function Index() {
                         userId,
```
Comment (solderq35) [inline app/content/profile/index.js:353]: Should we be doing try / catch here? Maybe make a handler function? Also goes for similar calls to handleImageUpload in team index page (team image upload).

The hooks/imageUpload.js file (where the handleImageUpload function is defined) also needs checking to see if it has enough try catches.

**Review by solderq35 — CHANGES_REQUESTED** [review]

**Comment by solderq35** [comment]
> Got an error about "Rendered fewer hooks than expected" when removing a player once, but did not happened again on a new user. All other promote, demote, and ban features work as intended

I think I fixed this now with https://github.com/Golf-Drill-Challenge-App/Golf-App/pull/239/commits/823b1224612d035958041a9e067600584dabe885, but needs more testing.

I was able to reproduce the "fewer hooks" bug after trying to remove a user, but not after my changes. Basically, I just redefined invalidateMultipleKeys function as async, and did `await invalidateMultipleKeys` everywhere.

I suspect that the app was trying to do `navigation.goBack()` before all keys were invalidated, which caused the error.

Although `queryClient.invalidateQueries` is not inherently async (why I don't bother doing `await queryclient`), doing an async / await ensured the invalidation and navigation happened in the right order (so it's more of a Promise than await I guess).

File: app/content/team/index.js [hunk 2]
```diff
@@ -150,7 +155,7 @@ function Index() {
       await updateDoc(doc(db, "teams", currentTeamId), {
         name: newName,
       });
-      invalidateMultipleKeys(queryClient, [["teamInfo"]]);
+      await invalidateMultipleKeys(queryClient, [["teamInfo"]]);
```
Comment (solderq35) [inline app/content/team/index.js:158]: Do we need try / catch / showDialog here or in handleUpdate function?

File: app/content/team/index.js [hunk 3]
```diff
@@ -330,10 +342,12 @@ function Index() {
                           currentTeamId,
                           teamRef,
                         );
-                        invalidateMultipleKeys(queryClient, [["teamInfo"]]);
+                        await invalidateMultipleKeys(queryClient, [
```
Comment (solderq35) [inline app/content/team/index.js:345]: this might need try / catch / showdialog, see other comment about handleimageupload on this pr

**Review by solderq35 — CHANGES_REQUESTED** [review]

**Pull request merged**